mvpa2/algorithms/statis.py

```diff
@@ -8,93 +8,111 @@
 ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
 """Transformation of individual feature spaces into a compromise common space
 
-The :class:`Statis` class in this module implements an algorithm
-published in Abdi, H., Williams, L.J., Valentin, D., & Bennani-Dosse, M. (2012). STATIS
-and DISTATIS: Optimum multi-table principal component analysis and three way
-metric multidimensional scaling. Wiley Interdisciplinary Reviews:
-Computational Statistics, 4, 124-167.
+See :class:`Statis` for more information.
 
 """
 
 import numpy as np
+from mvpa2.base.constraints import EnsureInt, EnsureRange
 from mvpa2.base.types import is_datasetlike
 from mvpa2.base.state import ClassWithCollections
 from mvpa2.mappers.zscore import ZScoreMapper
 from mvpa2.mappers.staticprojection import StaticProjectionMapper
 from mvpa2.base.dataset import hstack
+from mvpa2.base.param import Parameter
+
+from mvpa2.base import warning
+if __debug__:
+    from mvpa2.base import debug
+
+# make only main utility visible outside
+
+__all__ = ['Statis']
 
 def normalize_ds(ds):
-    """
-    Normalizes input dataset by
-    1) zscoring each column
-    2) making sum of squares to one (Frobenius norm = 1)
-    :param ds:
-    :return ds:
+    """Normalize input dataset for Statis
+
+    1) zscore each column
+    2) make sum of squares to one (Frobenius norm = 1)
     """
     if ds.samples.dtype == int:
         ds.samples = ds.samples.astype('float')
     zmapper = ZScoreMapper(chunks_attr=None)
     zmapper.train(ds)
     ds = zmapper.forward(ds)
-    ds.samples = ds.samples/np.sqrt(((ds.samples**2).sum()))
+    ds.samples /= np.sqrt(((ds.samples**2).sum()))
```

Comment:
because of what standardization does -- makes std of a signal 1 and mean 0;
```
In [9]: import numpy as np; d=np.random.rand(1235)+1.2; print(np.linalg.norm(d)); d-=np.mean(d); d/=np.std(d); print(np.linalg.norm(d)); d/=np.sqrt(len(d)); print(np.linalg.norm(d));
60.2847926739
35.1425667816
1.0
```